app/controllers/users/approvals_controller.rb

```diff
@@ -1,10 +1,14 @@
 class Users::ApprovalsController < ApplicationController
 
-  before_action :authenticate_user!
+  before_action :authenticate_user! 
+  before_action :check_user, only: :index
 
   def index
     @approved_devs = current_user.approved_developers
     @pending_approvals = current_user.pending_approvals
+    if @pending_approvals.length == 0 && params[:redirect]
+      redirect_to params[:redirect]
```

Comment:
What we could do is rather than passing this in the URL, allow the Developer to specify a redirect URL when they sign up (/ on their profile).
This could then be something like `redirect_to @developer.redirect_path`
Adding a little security